third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -650,6 +656,26 @@ func resourceSqlDatabaseInstanceCreate(d *schema.ResourceData, meta interface{})
 
 	d.Set("name", name)
 
+	// Before we create the instance, check if at least 1 service connection exists for private SQL Instances.
```

Comment:
This change is completely unrelated to the deletion protection one. In the future, changes like these two should be in separate PRs so they can be easily reviewed independently.